.github/workflows/build.yml

```diff
@@ -0,0 +1,54 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: ${{ matrix.os }} / GHC ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os:
+          - ubuntu-16.04 # https://launchpad.net/~hvr/+archive/ubuntu/ghc?field.series_filter=xenial
+        ghc:
+          - 7.0.4
+          - 7.2.2
+          - 7.4.2
+          - 7.6.3
+          - 7.8.4
+          - 7.10.3
+          - 8.0.2
+          - 8.2.2
+          - 8.4.4
+          - 8.6.5
+          - 8.8.4
+          - 8.10.3
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1
+        with:
+          ghc-version: ${{ matrix.ghc }}
+
+      - run: |
+          mkdir -p "$HOME/.local/bin"
+          curl -L https://github.com/haskell-compat/base-compat/releases/download/typediff-0.1.4/typediff > "$HOME/.local/bin/typediff"
+          chmod a+x "$HOME/.local/bin/typediff"
+          echo "$HOME/.local/bin" >> "$GITHUB_PATH"
+
+      - uses: sol/run-haskell-tests@v1
```

Comment:
I've never used this test runner before, so I'm unclear if it actually does everything that `haskell-ci` does. For example, does it run Haddocks? Does it run `cabal check` (which is necessary to pass if you want to upload these to Hackage)?

Reply:
It only runs the tests (https://github.com/sol/run-haskell-tests/blob/main/action.yml):
```yaml
runs:
  using: composite
  steps:
    - shell: bash
      run: cabal update

    - shell: bash
      run: cabal configure --enable-tests --enable-benchmarks # NOTE: --enable-tests is necessary due to https://github.com/haskell/cabal/issues/5079

    - shell: bash
      run: cabal build all

    - shell: bash
      run: cabal test all --test-show-details=direct
      env:
        HSPEC_OPTIONS: --color
```

Anything else you would need to add on top of that.

To be clear, this is just a suggestion.  Feel free to cherry-pick what is useful or stick with `haskell-ci`.

Reply:
Haddocks and `cabal check` are the only things that come to mind, so I'd be content to add those.